### ISSUE TITLE
Revisiting no-whitespace-before-property` false positive #5167

### ISSUE THREAD

**Issue by txwizard** [eslint/eslint#10104]
**Tell us about your environment**

* **ESLint Version:** 4.10.0
* **Node Version:** 8.10.0
* **npm Version:** 5.6.0

**What parser (default, Babel-ESLint, etc.) are you using?** Default, I presume, since the CLI command that starts the server is **npm start**, executing against the contents of my development directory.

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

The installations of both Node and MPM are new to this workstation, and the only modules in the local repository, from which the React application launches, are the ones installed by the kit provided in the tutorial. If I understand correctly the examples cited in issue 5167, "no-whitespace-before-property` false positive", at  https://github.com/eslint/eslint/issues/5167, this rule shouldn't flag anything in my source.

The attached zip contains the entire package, excluding **node_modules**, which seems redundant, since that part is readily available.

The workstation is a H-P z240 Tower Workstation equipped with 32 GB of RAM and 2.5 TB of mass storage, split between a 512 GB SSD and a 2.0 TB 7200 RPM magnetic volume, all of which have ample free space. The operating system is Microsoft Windows 10 Professional, Build 1709, and it is fully patched through last week.

/tic_tac_toe/node_modules/eslint/conf/eslint-recommended.js

/**
 * @fileoverview Configuration applied when a user configuration extends from
 * eslint:recommended.
 * @author Nicholas C. Zakas
 */

"use strict";

/* eslint sort-keys: ["error", "asc"] */

module.exports = {
    rules: {
        "accessor-pairs": "off",
        "array-bracket-newline": "off",
        "array-bracket-spacing": "off",
        "array-callback-return": "off",
        "array-element-newline": "off",
        "arrow-body-style": "off",
        "arrow-parens": "off",
        "arrow-spacing": "off",
        "block-scoped-var": "off",
        "block-spacing": "off",
        "brace-style": "off",
        "callback-return": "off",
        camelcase: "off",
        "capitalized-comments": "off",
        "class-methods-use-this": "off",
        "comma-dangle": "off",
        "comma-spacing": "off",
        "comma-style": "off",
        complexity: "off",
        "computed-property-spacing": "off",
        "consistent-return": "off",
        "consistent-this": "off",
        "constructor-super": "error",
        curly: "off",
        "default-case": "off",
        "dot-location": "off",
        "dot-notation": "off",
        "eol-last": "off",
        eqeqeq: "off",
        "for-direction": "off",
        "func-call-spacing": "off",
        "func-name-matching": "off",
        "func-names": "off",
        "func-style": "off",
        "function-paren-newline": "off",
        "generator-star-spacing": "off",
        "getter-return": "off",
        "global-require": "off",
        "guard-for-in": "off",
        "handle-callback-err": "off",
        "id-blacklist": "off",
        "id-length": "off",
        "id-match": "off",
        indent: "off",
        "indent-legacy": "off",
        "init-declarations": "off",
        "jsx-quotes": "off",
        "key-spacing": "off",
        "keyword-spacing": "off",
        "line-comment-position": "off",
        "linebreak-style": "off",
        "lines-around-comment": "off",
        "lines-around-directive": "off",
        "lines-between-class-members": "off",
        "max-depth": "off",
        "max-len": "off",
        "max-lines": "off",
        "max-nested-callbacks": "off",
        "max-params": "off",
        "max-statements": "off",
        "max-statements-per-line": "off",
        "multiline-comment-style": "off",
        "multiline-ternary": "off",
        "new-cap": "off",
        "new-parens": "off",
        "newline-after-var": "off",
        "newline-before-return": "off",
        "newline-per-chained-call": "off",
        "no-alert": "off",
        "no-array-constructor": "off",
        "no-await-in-loop": "off",
        "no-bitwise": "off",
        "no-buffer-constructor": "off",
        "no-caller": "off",
        "no-case-declarations": "error",
        "no-catch-shadow": "off",
        "no-class-assign": "error",
        "no-compare-neg-zero": "error",
        "no-cond-assign": "error",
        "no-confusing-arrow": "off",
        "no-console": "error",
        "no-const-assign": "error",
        "no-constant-condition": "error",
        "no-continue": "off",
        "no-control-regex": "error",
        "no-debugger": "error",
        "no-delete-var": "error",
        "no-div-regex": "off",
        "no-dupe-args": "error",
        "no-dupe-class-members": "error",
        "no-dupe-keys": "error",
        "no-duplicate-case": "error",
        "no-duplicate-imports": "off",
        "no-else-return": "off",
        "no-empty": "error",
        "no-empty-character-class": "error",
        "no-empty-function": "off",
        "no-empty-pattern": "error",
        "no-eq-null": "off",
        "no-eval": "off",
        "no-ex-assign": "error",
        "no-extend-native": "off",
        "no-extra-bind": "off",
        "no-extra-boolean-cast": "error",
        "no-extra-label": "off",
        "no-extra-parens": "off",
        "no-extra-semi": "error",
        "no-fallthrough": "error",
        "no-floating-decimal": "off",
        "no-func-assign": "error",
        "no-global-assign": "error",
        "no-implicit-coercion": "off",
        "no-implicit-globals": "off",
        "no-implied-eval": "off",
        "no-inline-comments": "off",
        "no-inner-declarations": "error",
        "no-invalid-regexp": "error",
        "no-invalid-this": "off",
        "no-irregular-whitespace": "error",
        "no-iterator": "off",
        "no-label-var": "off",
        "no-labels": "off",
        "no-lone-blocks": "off",
        "no-lonely-if": "off",
        "no-loop-func": "off",
        "no-magic-numbers": "off",
        "no-mixed-operators": "off",
        "no-mixed-requires": "off",
        "no-mixed-spaces-and-tabs": "error",
        "no-multi-assign": "off",
        "no-multi-spaces": "off",
        "no-multi-str": "off",
        "no-multiple-empty-lines": "off",
        "no-native-reassign": "off",
        "no-negated-condition": "off",
        "no-negated-in-lhs": "off",
        "no-nested-ternary": "off",
        "no-new": "off",
        "no-new-func": "off",
        "no-new-object": "off",
        "no-new-require": "off",
        "no-new-symbol": "error",
        "no-new-wrappers": "off",
        "no-obj-calls": "error",
        "no-octal": "error",
        "no-octal-escape": "off",
        "no-param-reassign": "off",
        "no-path-concat": "off",
        "no-plusplus": "off",
        "no-process-env": "off",
        "no-process-exit": "off",
        "no-proto": "off",
        "no-prototype-builtins": "off",
        "no-redeclare": "error",
        "no-regex-spaces": "error",
        "no-restricted-globals": "off",
        "no-restricted-imports": "off",
        "no-restricted-modules": "off",
        "no-restricted-properties": "off",
        "no-restricted-syntax": "off",
        "no-return-assign": "off",
        "no-return-await": "off",
        "no-script-url": "off",
        "no-self-assign": "error",
        "no-self-compare": "off",
        "no-sequences": "off",
        "no-shadow": "off",
        "no-shadow-restricted-names": "off",
        "no-spaced-func": "off",
        "no-sparse-arrays": "error",
        "no-sync": "off",
        "no-tabs": "off",
        "no-template-curly-in-string": "off",
        "no-ternary": "off",
        "no-this-before-super": "error",
        "no-throw-literal": "off",
        "no-trailing-spaces": "off",
        "no-undef": "error",
        "no-undef-init": "off",
        "no-undefined": "off",
        "no-underscore-dangle": "off",
        "no-unexpected-multiline": "error",
        "no-unmodified-loop-condition": "off",
        "no-unneeded-ternary": "off",
        "no-unreachable": "error",
        "no-unsafe-finally": "error",
        "no-unsafe-negation": "error",
        "no-unused-expressions": "off",
        "no-unused-labels": "error",
        "no-unused-vars": "error",
        "no-use-before-define": "off",
        "no-useless-call": "off",
        "no-useless-computed-key": "off",
        "no-useless-concat": "off",
        "no-useless-constructor": "off",
        "no-useless-escape": "error",
        "no-useless-rename": "off",
        "no-useless-return": "off",
        "no-var": "off",
        "no-void": "off",
        "no-warning-comments": "off",
        "no-whitespace-before-property": "off",
        "no-with": "off",
        "nonblock-statement-body-position": "off",
        "object-curly-newline": "off",
        "object-curly-spacing": "off",
        "object-property-newline": "off",
        "object-shorthand": "off",
        "one-var": "off",
        "one-var-declaration-per-line": "off",
        "operator-assignment": "off",
        "operator-linebreak": "off",
        "padded-blocks": "off",
        "padding-line-between-statements": "off",
        "prefer-arrow-callback": "off",
        "prefer-const": "off",
        "prefer-destructuring": "off",
        "prefer-numeric-literals": "off",
        "prefer-promise-reject-errors": "off",
        "prefer-reflect": "off",
        "prefer-rest-params": "off",
        "prefer-spread": "off",
        "prefer-template": "off",
        "quote-props": "off",
        quotes: "off",
        radix: "off",
        "require-await": "off",
        "require-jsdoc": "off",
        "require-yield": "error",
        "rest-spread-spacing": "off",
        semi: "off",
        "semi-spacing": "off",
        "semi-style": "off",
        "sort-imports": "off",
        "sort-keys": "off",
        "sort-vars": "off",
        "space-before-blocks": "off",
        "space-before-function-paren": "off",
        "space-in-parens": "off",
        "space-infix-ops": "off",
        "space-unary-ops": "off",
        "spaced-comment": "off",
        strict: "off",
        "switch-colon-spacing": "off",
        "symbol-description": "off",
        "template-curly-spacing": "off",
        "template-tag-spacing": "off",
        "unicode-bom": "off",
        "use-isnan": "error",
        "valid-jsdoc": "off",
        "valid-typeof": "error",
        "vars-on-top": "off",
        "wrap-iife": "off",
        "wrap-regex": "off",
        "yield-star-spacing": "off",
        yoda: "off"
    }
};
</details>

I was working through the React/JS tutorial, as it appears at https://reactjs.org/tutorial/tutorial.html, running a local Web server as suggested therein, so that I could use my own editor to modify the code. As I was about to shut down the server, I noticed the following series of messages.
[tic_tac_toe_20180319_180355.zip](https://github.com/eslint/eslint/files/1827281/tic_tac_toe_20180319_180355.zip)

https://github.com/eslint/eslint/issues/5167

```js

```

F:\Praesidium\Armatus_Admin_EventRegistration\React\tic_tac_toe>npm start

The package supplied by Facebook executes eslint; I haven't bothered to chase down the command. However, since I'm running an unmodified distribution from Facebook, you probably already know the command, or can find it. I've copied the lint rules into this issue; see above.

**What did you expect to happen?**
In the course of working through the tutorial, I inserted spaces around square brackets, as is my usual practice, in the interest of making the code easier for me to read. Since I hadn't seen any messages on previous runs, before I began to beautify my code, I was taken by surprise by them.

**What actually happened? Please include the actual, raw output from ESLint.**

Compiling...
Compiled with warnings.

./src/index.js
  Line 39:   Unexpected whitespace before property i  no-whitespace-before-property
  Line 48:   Unexpected whitespace before property i  no-whitespace-before-property
  Line 140:  Unexpected whitespace before property i  no-whitespace-before-property
  Line 142:  Unexpected whitespace before property a  no-whitespace-before-property
  Line 142:  Unexpected whitespace before property a  no-whitespace-before-property
  Line 142:  Unexpected whitespace before property b  no-whitespace-before-property
  Line 142:  Unexpected whitespace before property a  no-whitespace-before-property
  Line 142:  Unexpected whitespace before property c  no-whitespace-before-property
  Line 143:  Unexpected whitespace before property a  no-whitespace-before-property

Search for the keywords to learn more about each warning.
To ignore, add // eslint-disable-next-line to the line before.

The above is repeated for each time the package was recompiled in response to changes detected by the NPM file system monitor.

Did I misunderstand the explanation in issue 5167, or should it be reopened?

**Comment by platinumazure** [comment]
Hi @txwizard, thanks for the issue.

Could you please include your actual code and configuration files? (Even if your config file just says something like `{ "extends": "eslint:recommended" }`, that's helpful to know.)

Unfortunately, ESLint is maintained entirely by volunteers and we don't really have time to look through tutorials or long explanations. It really helps if we can just see the code you're running ESLint on (albeit through create-react-app I guess?), so we can see what "Line 142" actually refers to without having to dig in a zip file.

Thanks in advance for helping us help you!

**Comment by not-an-aardvark** [comment]
Closing because not enough information was provided to triage this issue.
